content/plugins/notes/kubernetes.md

```diff
@@ -0,0 +1,207 @@
+    clusterLocatorMethods:
+        - 'config'
+    clusters:
+        - url: http://127.0.0.1:9999
+          name: minikube
+          authProvider: 'serviceAccount'
+          serviceAccountToken:
+              $env: K8S_MINIKUBE_TOKEN
+        - url: http://127.0.0.2:9999
+          name: gke-cluster-1
+          authProvider: 'google'
+```
+
+Here is the list of fields in detail:
+
+-   `serviceLocatorMethod`
+
+    Configures how to determine which clusters a component is running in.
+
+    **Valid values:**
+
+    -   `multiTenant` - This configuration assumes that all components run on all the
+        provided clusters.
+
+-   `clusterLocatorMethods`
+
+    An array used to determine where to retrieve **cluster configuration** from.
+
+    **Valid values:**
+
+    -   `config` - This will read cluster information from your app-config.
+
+-   `clusters`
+
+    Used by the `config` cluster locator method to construct Kubernetes clients.
+
+    -   `url`
+
+        The **base URL** to the Kubernetes control plane.
+
+        You can find it using the `kubectl cluster-info` command and taking the "Kubernetes master" value.  
+        The output should be similar to this:
+
+        ```
+        Kubernetes master is running at ...
+        ```
+
+    -   `name`
+
+        A name to **represent this cluster**, this must be unique within the `clusters`
+        array. This value will be displayed to the final User in the Service Catalog Kubernetes plugin.
+
+    -   `authProvider`
+
+        Determines how the Kubernetes client authenticates with the Kubernetes
+        cluster.
+
+        **Valid values:**
+
+        -   `serviceAccount` - Use of a Kubernetes [service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to access the Kubernetes API.  
+            **NOTE:** when this is used the `serviceAccountToken` field should also be set (see bellow).
+
+        -   `google` - Use of a user's Google auth token from the [Google auth plugin](https://backstage.io/docs/auth/) to access the Kubernetes API.
+
+    -   `serviceAccountToken` - _optional_
+
+        The Kubernetes service account token to be used when `authProvider` is set to `serviceAccount`.
+
+### Using RBAC Authorization
+
+The current RBAC permissions required are read-only cluster wide, for the
+following objects:
+
+-   pods
+-   services
+-   configmaps
+-   deployments
+-   replicasets
+-   horizontalpodautoscalers
+-   ingresses
+
+An example of a Role to grant read access to pods:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+    namespace: default
+    name: pod-reader
+rules:
+    - apiGroups: ['']
+      resources: ['pods']
+      verbs: ['get', 'watch', 'list']
+```
+
+## Surfacing your Kubernetes components as part of an entity
+
+There are two ways to surface your Kubernetes components as part of an entity.  
+**NOTE:** The label selector takes precedence over the annotation/service id.
```

Comment:
Is there a clear reason why I would use one method or the other? If so, we should add that here.

Reply:
I don't think there's a clear reason why one method should be used over the other. I just know that in the Backstage repository, examples use the annotation/service id.